dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/readerOMC.py

```diff
@@ -424,62 +424,184 @@ def read_info_json(self):
+                types.extend(child.get_types())
+            if self.type != None and len(self.children) == 0:
+                types.append(self.type)
+            return types
+
+        def get_diff_vars(self, diff_var):
+            diff_var.append(self.diff_var)
+            for child in self.children:
+                child.get_diff_vars(diff_var)
+
+        def propagate_differential(self):
+            for child in self.children:
+                if self.type != None and self.type == DIFFERENTIAL:
+                    child.force_differential()
+                else:
+                    child.propagate_differential()
+
+
+        def force_differential(self):
+            if self.type != None:
+                self.type = DIFFERENTIAL
+            for child in self.children:
+                child.force_differential()
+
+    def create_nodes_for_min_and_max(self, root):
+        if len(root.children) == 0:
+            nodes_to_add = 0
+            if "min" in root.eq:
+                nodes_to_add+= root.eq.count("min(")
+            if "max" in root.eq:
+                nodes_to_add+= root.eq.count("max(")
+            for _ in range(0, nodes_to_add):
+                node = self.Node()
+                node.eq = "MIN/MAX"
+                node.type = root.type
+                root.children.append(node)
+                node2 = self.Node()
+                node2.eq = "MIN/MAX"
+                node2.type = root.type
+                root.children.append(node2)
+        else:
+            for child in root.children:
+                self.create_nodes_for_min_and_max(child)
+
+    def analyse_eq_expr(self, eq, root):
+        der_var_ptrn = re.compile(r'der\s*\((?P<var>[\(\S\) ]+?)\)')
+        splitted_eq = eq.replace('[u\'', '').split(" ")
+        idx = 0
+        while idx < len(splitted_eq):
+            expr = splitted_eq[idx]
+            if (len(expr) == 0):
+                idx += 1
+                continue
+            if expr.startswith("(if") or expr.startswith("-(if") or expr.startswith("- (if"):
+                nb_parenthesis = 1
+                sub_eq= expr.replace("(","", 1)
+                idx+=1
+                while nb_parenthesis > 0:
+                    if "(" in splitted_eq[idx]:
+                        nb_parenthesis+=splitted_eq[idx].count("(")
+                    if ")" in splitted_eq[idx]:
+                        nb_parenthesis-=splitted_eq[idx].count(")")
+                    sub_eq+= " " + splitted_eq[idx]
+                    idx+=1
+                self.analyse_eq_expr(sub_eq[:-1], root)
+            elif expr.startswith("if") or expr.startswith("-if"):
+                idx = self.remove_condition(idx, splitted_eq)
+                node = self.Node()
+                node.eq = "IF"
+                root.children.append(node)
+                (final_expr, idx) = self.find_value(idx, splitted_eq)
+                self.analyse_eq_expr(final_expr, node)
+            elif expr.startswith("else"):
```

Comment:
No other possibilities there too to support, such as "- else" or anything similar?

Reply:
no, the syntax is var1 -(if cond then var2 else -var3) so it is not possible to have a minus in front of a else